src/renderer/components/kubeconfig-dialog/open.injectable.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import type React from "react";
+import loggerInjectable from "../../../common/logger.injectable";
+import showCheckedErrorNotificationInjectable from "../notifications/show-checked-error.injectable";
+import kubeconfigDialogStateInjectable from "./state.injectable";
+
+export interface OpenKubeconfigDialogArgs {
+  title?: React.ReactNode;
+  loader: () => Promise<string>;
+}
+
+export type OpenKubeconfigDialog = (openArgs: OpenKubeconfigDialogArgs) => void;
+
+const openKubeconfigDialogInjectable = getInjectable({
+  id: "open-kubeconfig-dialog",
+  instantiate: (di): OpenKubeconfigDialog => {
+    const state = di.inject(kubeconfigDialogStateInjectable);
+    const showCheckedErrorNotification = di.inject(showCheckedErrorNotificationInjectable);
+    const logger = di.inject(loggerInjectable);
+
+    return ({ title, loader }) => {
+      (async () => {
+        try {
+          const config = await loader();
+
+          state.set({ title, config });
+        } catch (error) {
+          showCheckedErrorNotification(error, "Failed to retrive config for dialog");
+          logger.warn("[KUBEOCONFIG-DIALOG]: failed to retrived config for dialog", error);
```

Comment:
Orphan promise is not required? Such are pretty difficult to unit-test, and should not be used without a very good cause.

Reply:
What do you mean by "not required"? This is more explicit in the fact that the loader has to run before the dialog is open since previously the loader was run in the component itself.

Reply:
I mean avoidable. To avoid orphan promises whenever possible has served pretty well thus far :)

Reply:
Resolved for being pretty trivial here.